src/js/app/modules/users/controllers/editUsers.controller.js

```diff
@@ -0,0 +1,137 @@
+            vm.seniorityApi = data;
+        });
+        ProjectFactory.query(function (data){
+            vm.projectApi = data;
+        });
+        AfpFactory.query(function (data){
+            vm.afpApi = data;
+        });
+        EpsFactory.query(function (data){
+            vm.epsApi = data;
+        });
+        SkillsFactory.query(function (data){
+            vm.skillsApi = data;
+        });
+
+        vm.extractSkills = function (){
+            //TODO: Convert this in a reusable component
+            vm.postObject.skill = [];
+            for (var property in vm.userInfo.skills){
+                if(vm.userInfo.skills[property] == true){
+                    vm.postObject.skill.push(property);
+                }
+            }
+        };
+
+        vm.validateFormData = function (){
+            // TODO: Convert this to a reusable component
+            // instead of a controller method
+
+            var userForm = $scope.editUsersForm;
+            vm.postObject.skill = [];
+            for (var key in userForm) {
+                var elem = userForm[key];
+
+                // Look for changed fields
+                if (typeof(elem) == 'object' && elem.hasOwnProperty('$dirty') && elem.$dirty === true) {
+                    var modelValue = null;
+                    // Validation fo Skills model
+                    var skillModel = elem.$name.substring(0,6);
+                    if (skillModel == 'skills'){
+                        vm.extractSkills();
+                    }else {
+                        // On these fields save just the id and not the whole object for this Select lists
+                        var elementsName = [
+                            'socialStratum', 'documentType', 'birthPlace', 'nationality', 'maritalStatus', 'scholarship',
+                            'rh', 'seniority', 'project', 'afp', 'eps'
+                        ];
+                        if (elementsName.indexOf(elem.$name) >= 0){
+                            modelValue = elem.$modelValue.id;
+                        }else{
+                            modelValue = elem.$modelValue;
+                        }
+
+                        vm.postObject[elem.$name] = modelValue;
+                    }
+
+                }
+            }
+            if (vm.postObject.skill.length == 0) {vm.postObject.skill = null;}
+
+            UsersFactory.update(vm.postObject, {id: $routeParams.id}, function (response){
+                vm.postStatus = response;
+                vm.postStatus.message = "User successfully updated";
+                vm.postStatus.error = false;
+            },function (error){
+                vm.postStatus = error;
+                vm.postStatus.message = "Error " + error.status + " " + error.statusText + " | Message: " + error.data.summary;
+                vm.postStatus.error = true;
+                vm.postStatus.postedForm = true;
+            });
+
+            return true;
+        };
+    }
```

Comment:
If the process get an error in any step are managing exceptions and tx to do rollback ?